### ISSUE TITLE
Toolbar List/Grid mode getting cached with Turpentine Varnish

### ISSUE THREAD

**Issue by anithabolisetty** [nexcess/magento-turpentine#1179]
Hi,
Please find the screenshot link below and we are facing issue with toolbar selection modes List and Grid modes,

http://prnt.sc/ay8f1p

We need to utilize the varnish in category pages and at same time we need the view mode selection to be remembered and should work like how default Magento works, Main issue which we are trying to fix is we are unable to get the mode selected in Magento session when varnish is enabled, without varnish everything is working fine,

Note :- We need a way that this needs to be working like how default magento works at same time the category page needs to be using Turpentine Varnish

thanks

**Comment by miguelbalparda** [comment]
What have you tried so far?

**Comment by anithabolisetty** [comment]
HI, 
We have tried saving the modes in magento session and also magento registry keys, but none of those worked out.

As per our analysis, the registry keys and also magento session variables are getting cached,

thanks

**Comment by aricwatson** [comment]
It looks like what's happening is the category items are being cached in the general page cache. So, whichever state (grid or list) was active *when that URL was cached* remains cached for that URL.

I haven't had a chance to test it, but you should be able to fix this by adding an ESI policy for that block, setting access to `private` and scope to `page` - this should result in a separate version of this cached for each user and page.

**Comment by anithabolisetty** [comment]
thanks for your help, i have tried the same,  added access to private and scope to page for block,
it is working when user revisit to new pages, but the old page is still cached with those page specific changes
like for example, first opened  page A and made changes to toolbar and  clicked  page B, the settings of Page A being displayed in Page B, and again made few changes to toolbar in Page B , when  clicked again   page A then it is displaying the Page A toolbar settings but not page B toolbar settings 

thanks.

**Comment by miguelbalparda** [comment]
@anithabolisetty have you flushed your cache?

**Comment by anithabolisetty** [comment]
yes i have flushed cache after applying the changes but still the issue exists.

thanks.
